dbsc/workflow/rules/mrtpipelines.smk

```diff
@@ -364,6 +371,18 @@ rule tckgen:
     params:
         step=config["step"],
         sl=config["sl_count"],
+        tmp_dir=bids_tractography_out(
```

Comment:
I think these should probably be `resources`, otherwise the app will re-run if the `tmp_dir` changes, which I don't think we want.

Reply:
Yeah, I thought about this too but there were 2 concerns:
1. Whether it would pick up `$SLURM_TMPDIR` correctly (this was also a concern for putting this outside of rule and assigning it to a variable).
2. In this case, it should only re-run if it can't find the output in the final destination? It's okay if it reruns otherwise since it is likely intended (mrtrix errors out if you try to run it and it already exists in the expected location).

Reply:
1. Just so I'm clear, the concern is that it will evaluate `$SLURM_TMPDIR` on the snakemake host rather than the node running the job? Are `params` evaluated differently from `resources` to fix that? I'll note that if you use `resources.tmpdir`, [it definitely is evaluated on the node running the job](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#standard-resources).
2. A changed param automatically triggers a re-run even if the output is already in the final destination (Snakemake expects that the changed param will change the contents of the output). I think in this case it would delete whatever's already there and rerun the exact same command.

Reply:
1. Ahh okay, I did not know that (about evaluating on the node). I was more concerned about getting this running over the holidays given the time constraints at the time. Would need to point `resources.tmpdir` to the correct end destination then I guess.
2. I see, okay then would be ideal to be able to sort this out.

Reply:
Yeah so I hadn't considered 1 but I don't _think_ switching from `params` to `resources` should change the behaviour from that perspective. From the 2 perspective, it should be as simple (if a little labour intensive) as moving all the `tmp_*` params to the `resources` section and updating the `run`/`shell` commands appropriately.